File: src/main/java/cn/chahuyun/authorize/entity/UserPermissionInfo.java
package cn.chahuyun.authorize.entity;

import cn.chahuyun.authorize.HuYanAuthorize;
import cn.chahuyun.authorize.utils.HibernateUtil;
import jakarta.persistence.*;
import lombok.Data;

/**
 * 用户信息<p>
 *
 * @author Moyuyanli
 * @date 2023/1/3 9:04
 */
@Data
@Entity
@Table(name = "UserPermissionInfo")
public class UserPermissionInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    /**
     * 所属bot
     */
    private long bot;
    /**
     * 群号id
     */
    private long groupId;
    /**
     * 对象id
     * 可以是用户qq
     * 可以是群号
     */
    private long qq;
    /**
     * 是否全局
     */
    private boolean global;
    /**
     * 权限code
     */
    private String code;

    /**
     * 权限信息
     */
    @Transient
    private PermissionInfo permissionInfo;

    public UserPermissionInfo() {
    }

    public UserPermissionInfo(long qq, long groupId, long bot, boolean global, String code) {
        this.qq = qq;
        this.bot = bot;
        this.groupId = groupId;
        this.global = global;
        this.code = code;
    }

    /**
     * 保存
     *
     * @return true 成功
     */
    public boolean save() {
        try {
            HibernateUtil.factory.fromTransaction(session -> session.merge(this));
            return true;
        } catch (Exception e) {
            HuYanAuthorize.LOGGER.error("群成员基本权限添加失败:", e);
            return false;
        }
    }

    /**
     * 获取权限信息
     *
     * @return 权限信息
     */
    public PermissionInfo getPermissionInfo() {
        return HibernateUtil.factory.fromSession(session -> session.createQuery("from PermissionInfo as prem where prem.code = '" + code + "'", PermissionInfo.class).getSingleResult());
    }
}

File: src/main/kotlin/cn/chahuyun/authorize/entity/UserPermissionInfo.kt
package cn.chahuyun.authorize.entity

import cn.chahuyun.authorize.constant.UserType
import jakarta.persistence.*


@Entity(name = "auth_user_permission")
@Table(name = "auth_user_permission")
data class UserPermissionInfo(
    /**
     * id
     */
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Int? = null,
    /**
     *  bot
     */
    var bot: Long? = null,
    /**
     * 群号
     */
    var groupId: Long? = null,
    /**
     * qq
     */
    var qq: Long? = null,
    /**
     * 用户类型
     */
    var type: UserType = UserType.GLOBAL,
    /**
     * 权限code
     */
    var code: String? = null,
) {

}